src/runtime/components/Popover.vue

```diff
@@ -47,14 +48,23 @@ export interface PopoverEmits extends PopoverRootEmits {
   'close:prevent': []
 }
 
-export interface PopoverSlots {
-  default(props: { open: boolean }): any
-  content(props?: {}): any
-  anchor(props?: {}): any
+export interface PopoverSlots<M extends PopoverMode = PopoverMode> {
+  default(props: {
+    open: boolean
+    close: M extends 'click' ? (() => void) : never
+  }): any
+
+  content(props: {
+    close: M extends 'click' ? (() => void) : never
+  }): any
+
+  anchor(props: {
+    close: M extends 'click' ? (() => void) : never
+  }): any
 }
 </script>
 
-<script setup lang="ts">
+<script setup lang="ts" generic="M extends PopoverMode = PopoverMode">
```

Comment:
There should not be a default in the generic declaration:
```suggestion
<script setup lang="ts" generic="M extends PopoverMode">
```